src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs

```diff
@@ -143,13 +143,17 @@ internal static Exception CreateWrappedException(Exception exception, string hos
 
             static HttpRequestError DeduceError(Exception exception)
             {
-                // TODO: Deduce quic errors from QuicException.TransportErrorCode once https://github.com/dotnet/runtime/issues/87262 is implemented.
                 if (exception is AuthenticationException)
                 {
                     return HttpRequestError.SecureConnectionError;
                 }
 
-                if (exception is SocketException socketException && socketException.SocketErrorCode == SocketError.HostNotFound)
+                // Resolving a non-existent hostname often leads to EAI_AGAIN/TryAgain on Linux, indicating a non-authoritative failure, eg. timeout.
```

Comment:
Should we try to check `exception is HttpIOException` and `exception is HttpRequestException` in case the user was able to provide a better error code from their `ConnectCallback`?

Reply:
Forwarding or rewrapping (?) `HttpRequestException` makes sense. I'm skeptical if they should throw `HttpIOException` unless they are doing something crazy with an embedded http request in the callback. Do you think it's a realistic scenario we should care about?

Reply:
I guess it would be a bit odd. `ConnectionError` still makes sense as a fallback.

This also feels like something we should be fine with changing in the future since the user is the one providing the different value.